### PR TITLE
229 django forms

### PR DESCRIPTION
**Pull request by sebgrebe**
addresses #229 

Aim was to replace html for forms by django-crispy forms. Ideally, each form would just be displayed by {{ form | crispy }} in the html.

For a couple of forms, I could not do this because rendering specific fields adequately through Django-crispy-forms proved difficult. In those cases I rendered those fields through html and all others fields in this form through {{ form.FIELDNAME | as_crispy_field }}. Those forms were:
- edit_profile.html. Problematic field: image upload (looks terrible with django forms and I couldn't figure out a non-hacky way to fix that)
- edit_profile_cause_areas.html: Problematic fields: cause areas and giving pledges taken. In django forms, they display the id of the cause area or giving pledge, not their name. Probably fixable, but couldn't figure it out.
- edit_profile_career: Problematic field: Expertise. Same problem as in previous file. 
- login.html: Alert appeared above email and password. Had to render it through html to fix its position

Independent of whether this can be simplified further, future additional forms can be simply implemented through django-forms with this PR.

### CODE REVIEW

File: eahub/profiles/forms.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def save(self, commit=True):
 
     class Meta(CaseInsensitiveUsernameFieldCreationForm.Meta):
         model = User
-        fields = ['email']
+        fields = ['name', 'email']
```
Comment (taymonbeal) [inline eahub/profiles/forms.py]: Why this change? Does the signup flow still work properly after making it? I ask because this is a `ModelForm` and the `User` model doesn't have a `Name` field, so I'm not sure whether it's right or wrong.

File: eahub/config/build_settings.py [hunk 2]
```diff
@@ -21,3 +22,5 @@
 STATIC_ROOT = "/static/"
 STATIC_URL = "/static/"
 STATICFILES_STORAGE = "whitenoise.storage.CompressedManifestStaticFilesStorage"
+
+CRISPY_TEMPLATE_PACK = 'bootstrap4'
```
Comment (taymonbeal) [inline eahub/config/build_settings.py]: Add a comment `# django-crispy-forms` above this line, to follow convention.

**Comment by taymonbeal** [comment]
Note that #229 will need to remain open until the points that you've addressed here are solved (which I'll look into as soon as I get the chance). But this PR can be merged without that, once the above issues are taken care of.

File: eahub/profiles/forms.py (hunk 1 above)
Comment (sebgrebe) [reply eahub/profiles/forms.py]: This was necessary to make sure the name appears before the email on the form.

**Comment by sebgrebe** [comment]
Can we merge this?

**Comment by taymonbeal** [comment]
Yes, that's what it means for it to be approved.

**Comment by sebgrebe** [comment]
> Yes, that's what it means for it to be approved.

Thought the reviewer merges
